### ISSUE TITLE
Report parameters

### ISSUE THREAD

**Issue by AleCandido** [N3PDF/yadism#15]
**TL;DR**: default and/or fallbacks with report parameters or raising errors for anything, including missing arguments?

I'm wondering how to do this (actually if to do this).

In apfel once that defaults are applied, conflicts are solved and so on, it prints the parameters that is actually sing for both the evolution (not in this project) and the DIS observables (if you initialize also apfelDIS).

Discussing with @felixhekhorn this week he suggested that this feature may be not reimplemented, instead it's better to raise an error if there is any conflict at all.
I quite agree with Felix, errors are more pythonic than hidden fallback for sure, and I think that it is also better in general, but now I'm considering if including any default at all.

So **the question is**: if there is any input, any default of any kind, I think that would be better to report parameters used in some way (dumping on a file, printing as output, ...), otherwise we should raise an error also for any missing argument.
